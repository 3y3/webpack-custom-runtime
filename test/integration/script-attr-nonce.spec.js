'use strict';

const cct = require('./utils/custom-configurable-test');
const WebpackCustomRuntime = require('../../').CustomRuntimePlugin;

describe('ScriptAttr(Nonce)', function() {

    function cit(name, options, config) {
        cct(name, Object.assign({
            plugins: [
                new WebpackCustomRuntime({ behavior: '' }),
                new WebpackCustomRuntime.ScriptAttr.Nonce(options),
            ]
        }, config));
    }

    cit('should override script nonce by default', { value: 'Ya.nonce' });

    cit('should override script nonce from custom holder', { value: 'Ya.nonce' });

    cit('should not to override script nonce with empty holder', { value: '' });
});
