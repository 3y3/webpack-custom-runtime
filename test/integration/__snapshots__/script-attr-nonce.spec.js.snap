// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptAttr(Nonce) should not to override script nonce with empty holder: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(Nonce) should not to override script nonce with empty holder: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(Nonce) should not to override script nonce with empty holder: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(Nonce) should not to override script nonce with empty holder: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(Nonce) should not to override script nonce with empty holder: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce by default: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce by default: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce by default: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce by default: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce by default: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ -151,0 +151,4 @@
+/******/ 	  if (Ya.nonce) {
+/******/ 	    result.nonce = Ya.nonce;
+/******/ 	  }
+/******/
"
`;

exports[`ScriptAttr(Nonce) should override script nonce from custom holder: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce from custom holder: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce from custom holder: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce from custom holder: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(Nonce) should override script nonce from custom holder: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ -151,0 +151,4 @@
+/******/ 	  if (Ya.nonce) {
+/******/ 	    result.nonce = Ya.nonce;
+/******/ 	  }
+/******/
"
`;
